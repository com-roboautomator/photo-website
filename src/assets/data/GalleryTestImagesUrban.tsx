import collection from "./ImageDataStructure";
import {v4 as uuid} from 'uuid'

export const urbanImages: collection[] = [
    {
        key: uuid(),
        url:
            'https://picsum.photos/id/101/300/300', // won't be required anymore
        titleImage: 4,    
        title: 'Buildings Collection',
        tagTitle: 'Architecture',
        tagColour: '#5f5f5f',
        index: 0,

        images: [
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1031/300/300',
                title: 'Skyscraper',
                index: 0,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1029/300/300',
                title: 'New York',
                index: 1,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1040/300/300',
                title: 'Castle',
                index: 2,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1047/300/300',
                title: 'Alley',
                index: 3,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/1067/300/300',
                title: 'City View',
                index: 4,
            },
            {
                key: uuid(),
                url: 'https://picsum.photos/id/164/300/300',
                title: 'Lake Side View',
                index: 5,
            },
        ]
    },

    {
        key: uuid(),
        url:
            'https://picsum.photos/id/1026/300/300',
        title: 'Transport Collection',
        tagTitle: 'Cityscapes',
        tagColour: '#7baab5',
        index: 1,
    },

    {
        key: uuid(),
        url:
            'https://picsum.photos/id/1040/300/300',
        title: 'Historic Building Collection',
        tagTitle: 'Architecture',
        tagColour: '#5f5f5f',
        index: 2,
    },

    {
        key: uuid(),
        url:
            'https://picsum.photos/id/1067/300/300',
        title: 'Cityscapes Collection',
        tagTitle: 'Cityscapes',
        tagColour: '#7baab5',
        index: 3,
    },
]
