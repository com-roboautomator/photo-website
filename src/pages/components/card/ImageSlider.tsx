import { createStyles, withStyles, WithStyles } from '@material-ui/core'
import React from 'react'
import CollectionCard from './CollectionCard'
import Arrow from './Arrow'
import Underscore from 'underscore'
import {v4} from 'uuid'

interface ImageSliderProps extends WithStyles<typeof styles> {
    data: { url: string; title: string; tagTitle?: string | undefined; tagColour?: string | undefined; index: number; }[]
}

class ImageSlider extends React.Component<ImageSliderProps, any> {
    constructor(props: ImageSliderProps) {
        super(props)
        this.handleWheelMovement = Underscore.throttle(this.handleWheelMovement, 300);
        this.state = {
            selected: 0,
            properties: this.props.data,
            property: this.props.data[0]
        }
    }

    next = () => {
        if (this.state.property.index === this.state.properties.length - 1) return
        const newIndex = this.state.property.index + 1
        this.setState({
            property: this.state.properties[newIndex],
            selected: newIndex,
        })
    }

    previous = () => {
        if (this.state.property.index === 0) return
        const newIndex = this.state.property.index - 1
        this.setState({
            property: this.state.properties[newIndex],
            selected: newIndex,
        })
    }

    handleWheelMovement = (e: React.WheelEvent<HTMLDivElement>) => {
        const movement = (e.deltaY) ? e.deltaY : e.deltaX
        if (movement > 0) {
            this.next()
        } else {
            this.previous()
        }
    }

    render() {
        const classes = this.props.classes
        const { properties, property } = this.state

        return (
            <div className={classes.container} >
                <div className={classes.button_wrapper}>
                    <div className={classes.arrow} >
                        <Arrow
                            disabled={property.index === 0}
                            onClick={this.previous}
                            orientation={'Left'}
                        />
                    </div>
                    <div className={classes.arrow}>
                        <Arrow
                            disabled={
                                property.index === properties.length - 1
                            }
                            onClick={this.next}
                            orientation={'Right'}
                        />
                    </div>
                </div>
                <div className={classes.card_slider} onWheel={(e) => { this.handleWheelMovement(e) }}>
                    <div
                        className={classes.card_slider_wrapper}
                        style={{
                            transform: `translateX(-${property.index * (100 / properties.length)
                                }%)`,
                        }}>
                        {properties.map(
                            (property: {
                                key: string
                                url: string
                                title: string
                                tagTitle: string
                                tagColour: string
                                index: number
                            }) => (
                                <CollectionCard
                                    key={v4()}
                                    coverSrc={property.url}
                                    title={property.title}
                                    selected={
                                        this.state.selected ===
                                        property.index
                                    }
                                    tagTitle={property.tagTitle}
                                    tagColour={property.tagColour}
                                />
                            )
                        )}
                    </div>
                </div>
            </div>
        )
    }
}

const styles = () =>
    createStyles({
        container: {

            pointerEvents: 'auto',
            position: 'relative',
            display: 'flex',
            justifyContent: 'center',
            overflowX: 'hidden',
        },
        card_slider: {
            pointerEvents: 'auto',
            position: 'relative',
            maxWidth: '370px',
            width: '370px',
            height: '270px',
        },
        button_wrapper: {
            zIndex: 99,
            pointerEvents: 'none',
            position: 'absolute',
            width: '100vw',
            paddingTop: '70px',
            paddingBottom: '70px',
            display: 'flex',
            justifyContent: 'space-between',
            alignSelf: 'center',
            background:
                'linear-gradient(to right, white, transparent 50%), linear-gradient(to left, white, transparent 50%)',
        },
        card_slider_wrapper: {
            pointerEvents: 'auto',
            display: 'flex',
            position: 'absolute',
            margin: '10px',
            transition: '300ms cubic-bezier(0.455, 0.03, 0.515, 0.955)',
        },

        arrow: {
            pointerEvents: 'auto',
            margin: '20px',
        },
    })

export default withStyles(styles)(ImageSlider)