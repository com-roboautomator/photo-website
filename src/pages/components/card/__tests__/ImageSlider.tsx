import { fireEvent, render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import ImageSlider from '../ImageSlider'
import { data } from '../../../../assets/data/TestData'

describe('Image Slider', () => {

    it('renders correctly when there are no items', () => {
        const ImageSliderRender = renderer
            .create(
                <ImageSlider data={data} />
            )
            .toJSON()

        expect(ImageSliderRender).toMatchSnapshot()
    })

    it('should increment image index on arrow button click', () => {

        const component = render(<ImageSlider data={data} />)

        expect(component).toBeTruthy();
        expect(component.getByTestId("Arrow-Button-Right")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Nature Collection")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Urban Collection")).toBeTruthy()

        expect(component.getByTestId("Collection-Card-Nature Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")
        expect(component.getByTestId("Collection-Card-Urban Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")

        fireEvent.click(component.getByTestId("Arrow-Button-Right"))

        expect(component.getByTestId("Collection-Card-Nature Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")
        expect(component.getByTestId("Collection-Card-Urban Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")

    })

    it('should decrement image index on arrow button click', () => {

        const component = render(<ImageSlider data={data} startingIndex={3} />)

        expect(component).toBeTruthy();
        expect(component.getByTestId("Arrow-Button-Left")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Mechanical Collection")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Animal Collection")).toBeTruthy()

        expect(component.getByTestId("Collection-Card-Mechanical Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")
        expect(component.getByTestId("Collection-Card-Animal Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")

        fireEvent.click(component.getByTestId("Arrow-Button-Left"))

        expect(component.getByTestId("Collection-Card-Mechanical Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")
        expect(component.getByTestId("Collection-Card-Animal Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")

    })

    it('should increment image index on positive wheel move', () => {

        const component = render(<ImageSlider data={data} />)

        expect(component.getByTestId("Image-Slider-Card-Slider")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Nature Collection")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Urban Collection")).toBeTruthy()

        expect(component.getByTestId("Collection-Card-Nature Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")
        expect(component.getByTestId("Collection-Card-Urban Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")

        fireEvent.wheel(component.getByTestId("Image-Slider-Card-Slider"), {
            deltaY: 30
        })

        expect(component.getByTestId("Collection-Card-Nature Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")
        expect(component.getByTestId("Collection-Card-Urban Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")

    })

    it('should decrement image index on negative wheel move', () => {

        const component = render(<ImageSlider data={data} startingIndex={3} />)

        expect(component.getByTestId("Image-Slider-Card-Slider")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Mechanical Collection")).toBeTruthy()
        expect(component.getByTestId("Collection-Card-Animal Collection")).toBeTruthy()

        expect(component.getByTestId("Collection-Card-Mechanical Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")
        expect(component.getByTestId("Collection-Card-Animal Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")

        fireEvent.wheel(component.getByTestId("Image-Slider-Card-Slider"), {
            deltaY: -30
        })

        expect(component.getByTestId("Collection-Card-Mechanical Collection").firstChild).toHaveClass("CollectionCard-spacing_unselected-7")
        expect(component.getByTestId("Collection-Card-Animal Collection").firstChild).toHaveClass("CollectionCard-spacing_selected-6")

    })

    it('should set left arrow button to disabled when index is 0', () => {

        const component = render(<ImageSlider data={data} startingIndex={0} />)

        expect(component).toBeTruthy();
        expect(component.getByTestId("Arrow-Button-Left")).toBeTruthy()
        expect(component.getByTestId("Arrow-Button-Right")).toBeTruthy()

        expect(component.getByTestId("Arrow-Button-Left")).toHaveAttribute("disabled")
        expect(component.getByTestId("Arrow-Button-Right")).not.toBeDisabled()

    })

    it('should set Right arrow button to disabled when index is max length', () => {

        const component = render(<ImageSlider data={data} startingIndex={data.length -1} />)

        expect(component).toBeTruthy();
        expect(component.getByTestId("Arrow-Button-Left")).toBeTruthy()
        expect(component.getByTestId("Arrow-Button-Right")).toBeTruthy()

        expect(component.getByTestId("Arrow-Button-Left")).not.toBeDisabled()
        expect(component.getByTestId("Arrow-Button-Right")).toHaveAttribute("disabled")

    })

    it('should render image slider correctly', () => {

        const component = render(<ImageSlider data={data} />)

        // TODO: test image slider rendering

    })

})