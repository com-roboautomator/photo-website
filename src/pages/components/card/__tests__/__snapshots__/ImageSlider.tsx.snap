// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Slider renders correctly when there are no items 1`] = `
<main
  className="ImageSlider-container-1"
>
  <div
    className="ImageSliderTitle-hidden-6"
    style={
      Object {
        "borderColor": undefined,
      }
    }
  >
    
  </div>
  <div
    className="ImageSlider-wrapper-2"
    data-testid="ImageSlider-Container"
  >
    <div
      className="ImageSliderButtonWrapper-button_wrapper-7"
      data-testid="ImageSlider-button-wrapper"
      style={
        Object {
          "background": "linear-gradient(to right, white, transparent 30%), linear-gradient(to left, white, transparent 30%)",
        }
      }
    >
      <div
        className="ImageSliderButtonWrapper-arrow-8"
      >
        <div
          className="Arrow-container-9"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
            data-testid="Arrow-Button-Left"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="nata"
                className="Arrow-left-10"
                data-testid="Arrow-Button-Image"
                src="arrow-disabled.png"
              />
            </span>
          </button>
        </div>
      </div>
      <div
        className="ImageSliderButtonWrapper-arrow-8"
        data-testid="ImageSlider-arrow"
      >
        <div
          className="Arrow-container-9"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="Arrow-Button-Right"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="nata"
                className="Arrow-right-11"
                data-testid="Arrow-Button-Image"
                src="arrow-enabled.png"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="ImageSlider-card_slider-3"
      data-testid="ImageSlider-Card-Slider"
      onWheel={[Function]}
      style={
        Object {
          "height": "undefinedpx",
        }
      }
    >
      <div
        className="ImageSlider-card_slider_wrapper-4"
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="CollectionCard-spacing_selected-12"
          data-testid="Home-Collection-Card-Nature Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/befee15c-c770-4450-b71f-f20f820b3549"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/1018/800"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Nature Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#47742c",
                      }
                    }
                  >
                    Nature
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Urban Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/16bdf4fe-692d-4e11-8fd4-9ea37e3fc34d"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/101/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Urban Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Urban
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Animal Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/2c615617-6372-4042-bce5-a97910a51b5a"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/1024/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Animal Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Nature
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Mechanical Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/3961f578-14ea-46aa-9b74-7f908fcd532b"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/111/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Mechanical Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#50b1d6",
                      }
                    }
                  >
                    Technology
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Party Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/37f92b19-2ea1-4742-b97b-81ed97e2e9c3"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/158/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Party Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Second Nature Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/6dcdba12-f48c-46cf-b1f6-07e8d1c5ed2b"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/1018/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Second Nature Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Second Urban Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/23d4a85c-b08f-4f8d-9e83-055af4b538e5"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/101/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Second Urban Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Second Animal Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/90563348-458c-43db-8754-ee003cd78886"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/1024/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Second Animal Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Second Mechanical Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/57215613-5204-400d-9bac-66eee9383903"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/111/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Second Mechanical Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Second Party Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/12478d6e-6609-4978-a947-585910168c43"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/158/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Second Party Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Wedding Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/00bc3915-d776-4ac7-815b-079901002683"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/1013/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Wedding Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-Sport Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/92160e93-b95b-4351-bfb7-72f55a7f8496"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/1023/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  Sport Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="CollectionCard-spacing_unselected-13"
          data-testid="Home-Collection-Card-People Collection"
        >
          <div
            className="CollectionCard-container-14"
            data-testid="Home-Collection-Card-Container"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="gallery/7967d259-0081-4f89-ac25-9174fc23b36c"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <img
                alt=""
                className="CollectionCardImage-layout-21"
                data-testid="Collection-Card-Image"
                height={200}
                src="https://picsum.photos/id/375/200"
                width="350"
              />
              <div
                className="CollectionCardText-text_wrapper-23"
                data-testid="Collection-Card-Text-Wrapper"
              >
                <img
                  alt=""
                  className="CollectionCardText-icon-24"
                  data-testid="Collection-Card-Image-Icon"
                  height={20}
                  src="collection-image-o.png"
                  width={20}
                />
                <p
                  className="CollectionCardText-text-25"
                  data-testid="Collection-Card-Title"
                >
                  People Collection
                </p>
                <div
                  className="CollectionCardText-tag-26"
                >
                  <div
                    className="HomeCollectionTag-layout-27"
                    data-testid="tag-layout"
                    style={
                      Object {
                        "background": "#323232",
                      }
                    }
                  >
                    Default
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
